modules/cactus-web/src/Accordion/Accordion.tsx

```diff
@@ -243,32 +364,81 @@ interface AccordionState {
 }
 
 const AccordionBase = (props: AccordionProps) => {
-  const { accordions, manageOpen, isManaged } = useContext(ProviderContext)
+  const {
+    accordions,
+    manageOpen,
+    manageFocus,
+    focusFirst,
+    focusLast,
+    isManaged,
+    managedAccordions,
+  } = useContext(ProviderContext)
   const id = useId()
+  const headerId = props.id ? `${props.id}-header` : `${id}-header`
+  const bodyId = props.id ? `${props.id}-body` : `${id}-body`
   let isOpen = false
+
   if (isManaged) {
     isOpen = accordions[id] === undefined ? false : accordions[id].open
   }
   const [state, setState] = useState<AccordionState>({ isOpen: isOpen })
 
+  useEffect(() => {
+    if (managedAccordions && managedAccordions.current !== null) {
+      managedAccordions.current.push(id)
+    }
+  }, [id, managedAccordions])
+
+  useEffect(() => {
+    return () => {
+      if (managedAccordions && managedAccordions.current !== null) {
+        const accordionIndex = managedAccordions.current.indexOf(id)
+        managedAccordions.current.splice(accordionIndex, 1)
+      }
+    }
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
+
   const toggleOpen = () => {
     if (isManaged) {
       if (manageOpen && typeof manageOpen === 'function') {
         manageOpen(id, !isOpen)
       }
     } else {
-      setState(state => ({ isOpen: !state.isOpen }))
+      setState(state => ({ ...state, isOpen: !state.isOpen }))
+    }
+  }
+
+  const shiftFocus = (offset: number) => {
+    if (isManaged && typeof manageFocus === 'function') {
+      manageFocus(id, offset)
+    }
+  }
+
+  const focusFirstAccordion = () => {
+    if (isManaged && typeof focusFirst === 'function') {
+      focusFirst()
+    }
+  }
+
+  const focusLastAccordion = () => {
+    if (isManaged && typeof focusLast === 'function') {
+      focusLast()
     }
   }
 
   const rest = omitMargins(props, 'width', 'maxWidth')
 
   return (
-    <div {...rest}>
+    <div id={props.id || id} data-id={id} {...rest}>
```

Comment:
```suggestion
    <div id={props.id || id} {...rest}>
```
see comment about querying id directly with querySelector